internal/controller/drplacementcontrol_controller.go

```diff
@@ -564,41 +564,37 @@ func GetDRClusters(ctx context.Context, client client.Client, drPolicy *rmn.DRPo
 	return drClusters, nil
 }
 
-// updateObjectMetadata updates drpc labels, annotations and finalizer, and also updates placementObj finalizer
-func (r DRPlacementControlReconciler) updateObjectMetadata(ctx context.Context,
-	drpc *rmn.DRPlacementControl, placementObj client.Object, log logr.Logger,
-) error {
-	var update bool
+// updateDRPCMetadata updates DRPC labels, annotations and finalizer only
+func (r DRPlacementControlReconciler) updateDRPCMetadata(
+	drpc *rmn.DRPlacementControl, placementObj client.Object,
+) (modified bool, err error) {
+	modified = false
 
-	update = rmnutil.AddLabel(drpc, rmnutil.OCMBackupLabelKey, rmnutil.OCMBackupLabelValue)
-	update = rmnutil.AddFinalizer(drpc, DRPCFinalizer) || update
+	if rmnutil.AddLabel(drpc, rmnutil.OCMBackupLabelKey, rmnutil.OCMBackupLabelValue) {
+		modified = true
+	}
+
+	if rmnutil.AddFinalizer(drpc, DRPCFinalizer) {
+		modified = true
+	}
 
 	vrgNamespace, err := selectVRGNamespace(r.Client, r.Log, drpc, placementObj)
```

Comment:
@OdedViner , maybe we can selectVRGNamespace and only then add finalizer and label?